Css/PreProcessor/Adapter/Less/Processor.php

```diff
@@ -93,29 +94,37 @@ public function processContent(AssetFile $asset)
      */
     protected function compileFile($filePath)
     {
-        $nodeCmdArgs = $this->getNodeArgsAsArray();
-        $lessCmdArgs = $this->getCompilerArgsAsArray();
-
-        $cmd = '%s ' . str_repeat(' %s', count($nodeCmdArgs)) . ' %s' . str_repeat(' %s', count($lessCmdArgs)) . ' %s';
-        $arguments = [];
-        $arguments[] = $this->getPathToNodeBinary();
-        $arguments = array_merge($arguments, $nodeCmdArgs);
-        $arguments[] = $this->getPathToLessCompiler();
-        $arguments = array_merge($arguments, $lessCmdArgs);
-        $arguments[] = $filePath;
-
-        // to log or not to log, that's the question
-        // also, it would be better to use the logger in the Shell class,
-        // since that one will contain the exact correct command, and not this sprintf version
-        // $logArguments = array_map('escapeshellarg', $arguments);
-        // $this->logger->debug('Less compilation command: `'
-        //     . sprintf($cmd, ...$logArguments)
-        //     . '`');
-
-        return $this->shell->execute(
-            $cmd,
-            $arguments
-        );
+        $process = new Process([
+            $this->getPathToNodeBinary(),
+            ...$this->getNodeArgsAsArray(),
+            $this->getPathToLessCompiler(),
+            ...$this->getCompilerArgsAsArray(),
+            $filePath,
+        ]);
+
+        $process->run();
+
+        if (!$process->isSuccessful()) {
```

Comment:
This bit is taken from the Process exception. I didn't want to expose a Symfony exception here as it's an implementation detail.